File: app-cpp/src/file-retrieval-server.cpp
#include <iostream>
#include <memory>

#include "IndexStore.hpp"
#include "ServerProcessingEngine.hpp"
#include "ServerAppInterface.hpp"

int main(int argc, char** argv)
{
    // TO-DO change the server port to a non-privileged port from argv[1]
    int serverPort = 1;
    // TO-DO read the number of worker threads from argv[2]
    int numWorkerThreads = 1;

    std::shared_ptr<IndexStore> store = std::make_shared<IndexStore>();
    std::shared_ptr<ServerProcessingEngine> engine = std::make_shared<ServerProcessingEngine>(store);
    std::shared_ptr<ServerAppInterface> interface = std::make_shared<ServerAppInterface>(engine);

    // create a thread that runs the ZMQ Proxy and create the worker threads
    engine->initialize(serverPort, numWorkerThreads);

    // read commands from the user
    interface->readCommands();

    return 0;
}

File: app-java/src/main/java/csc435/app/FileRetrievalServer.java
package csc435.app;

public class FileRetrievalServer
{
    public static void main(String[] args)
    {
        // TO-DO change server port to a non-privileged port from args[0]
        int serverPort = 1;
        // TO-DO read the number of worker threads from args[1]
        int numWorkerThreads = 1;

        IndexStore store = new IndexStore();
        ServerProcessingEngine engine = new ServerProcessingEngine(store);
        ServerAppInterface appInterface = new ServerAppInterface(engine);
        
        // create a thread that runs the ZMQ Proxy and create the worker threads
        engine.initialize(serverPort, numWorkerThreads);

        // read commands from the user
        appInterface.readCommands();
    }
}
